### DISCUSSION TITLE
Writing a plugin for a "event created/change" hook?

### DISCUSSION THREAD

**Discussion by TimDaub** [Kozea/Radicale#1439] [Ideas]
As title already says, I'm interested in adding a web hook plugin for whenever any change is made to an event in radicale (e.g. newly created, times rescheduled, etc.). On the website I've found that [plugins](https://radicale.org/3.0.html#documentation/plugins) are supported.

Where do you recommend hooking in to listen for event changes?

**Update**: I'm now trying to use radicale's caldav API and its etags to periodically check for changes on a particular resource.

**Comment by lkraav** [comment]
I could also use

* an email triggered
* RSS feed
* webhooks to Mattermost and/or Slack

I guess there was no movement here @TimDaub?

**Comment by TimDaub** [comment]
Actually, there are results. I'm using Radicale for a client's project that involves building a calendar.
Through iCalendar VALARMS, I'm now able in the app to send out RSVP emails and SMS. The full product is
not yet available and probably will not be open source. But individual components exist.

What you might find useful is:

- https://github.com/TimDaub/simple-caldav; A JavaScript client for Radicale
- https://github.com/TimDaub/http-sms-gateway; A small gateway software for sending SMS from an LTE stick
- https://github.com/TimDaub/at-bindings; A utility belt built around  `at` that lets you schedule a process's execution at a specific time
